js/node_shared/unix.js

```diff
@@ -0,0 +1,348 @@
+/* What is this file for?
+   We would like to transpile our OCaml code to JS, to make it runnable in
+   the browser and on Windows systems. The OCaml standard library, however,
+   has external C dependencies, some of which are unimplemented by the JSOO
+   transpiler.
+   In particular, the Unix module consists largely of external C calls which
+   are not defined symbols in the produced JS.
+   To make sure we can still call those C functions (which happens in things
+   such as Bos, Time_limit, etc), we reimplement those fundamental Unix calls
+   in Javascript.
+   Note that this doesn't need to be a perfect reimplementation, but we can
+   get by with "good enough" behavior.
+ */
+
+const { spawnSync } = require("node:child_process");
+const os = require("os");
+const fs = require("fs");
+const process = require("process");
+
+globalThis.spawnSync = spawnSync;
+globalThis.fs = fs;
+globalThis.os = os;
+globalThis.process = process;
+
+// mapping pid to its own process
+globalThis.pidToProcessTable = new Map();
+// mapping pid of parent to its children
+globalThis.pidToChildrenTable = new Map();
```

Comment:
I'm seeing `.set()`s and `.get()`s but no deletes for these. We should either clean up after ourselves or consider using a `WeakMap`

Reply:
I've made it such that when we `wait`, we now remove a pid's entry (as well as its children).